### PR TITLE
Switch to Config::Any mechanism

### LINKED ISSUE #36: Replace Viroverse::config with a Config::Any based static config file
The existing system is not good. Tagged "large" because it'll involve a bunch of replacements in places where the config is used, which may be hard to grep for in some cases? Most should be easy.)

### PR DESCRIPTION
**Pull request by fluffy-critter**
Fixes #36 

This replaces the `Viroverse::config` and `Viroverse::config_local` modules with a simpler configuration mechanism that allows the site to put one or more configuration files in the Viroverse root directory in any format supported by `Config::Any`, named `viroverse_local.*` or `viroverse.*` (with the former taking precedence over the latter). It still provides default configuration as before.

Since it does a hash-level merge, overriding part of a hash is now possible. For example, with this in a `viroverse_local.json`:

```json
{
    "storage": "/home/viroverse/vagrant-storage",
    "features" : {
        "censor_dates": true
    }
}
```

this will enable the `censor_dates` feature while keeping the other configured features as the defaults.

Currently there's no default configuration file checked in, and Viroverse will only use the defaults. I could certainly make a case for moving the non-computed defaults out into a checked-in `viroverse.json` or the like, however.

Also, all of the configuration lookups were replaced using a `sed` expression similar to:

```bash
find . -name '*.pm' -or -name '*.pl' -or -name '*.psgi' |
    xargs grep 'Viroverse::config::' |
    xargs gsed -i.bak 's/\$Viroverse::config::\([a-zA-Z_]\+\)/Viroverse::Config->conf->\{\1\}/g;s/use Viroverse::config/use Viroverse::Config/g'
```

and then manually fixing up cases I saw where the configuration value was being used as an interpolated string. It's quite possible I missed something!

### CODE REVIEW

**Comment by silby** [comment]
I would like the main commit on this branch, adding `Viroverse::Config`, to have some additional details in the commit message giving the rationale for the change.

File: devel/vagrant/viroverse_local.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
[the remark this answers is not in the document]
Comment (fluffy-critter) [reply devel/vagrant/viroverse_local.json]: Sure, makes sense. I happen to find json easy to edit but the lack of comments is a bit of a drag. I’ll probably go with `Config::General`

File: viroverse.conf [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+read_only_pw =
+
+# Read-write user
+read_write_user = viroverse_w
+read_write_pw =
+
+### VIROBLAST
+
+# These defaults won't work unless var/viroblast is a checkout of
+# viroverse-viroblast.git (or a symlink to one)
```
Comment (silby) [inline viroverse.conf]: We should replace this detail with a proper link to obtaining Viroblast (which for whatever reason is through [CoMotion](https://els.comotion.uw.edu/express_license_technologies/viroblast)).

Comment (fluffy-critter) [reply viroverse.conf]: k, I had no idea :)

Comment (silby) [reply viroverse.conf]: No reason you should've!

**Comment by silby** [comment]
Great!

**Pull request merged**